jdk/src/share/classes/sun/util/calendar/ZoneInfoFile.java

```diff
@@ -635,6 +635,16 @@ private static ZoneInfo getZoneInfo(String zoneId,
                     params[8] = 86400000;
                 }
 
+                // Note: known mismatching -> Africa/Cairo
+                // ZoneInfo :      startDayOfWeek=5     <= Thursday
+                //                 startTime=86400000   <= 24:00
+                // This:           startDayOfWeek=6     <= Friday
+                //                 startTime=0          <= 0:00
+                if (zoneId.equals("Africa/Cairo") &&
+                        params[7] == Calendar.FRIDAY && params[8] == 0) {
+                    params[7] = Calendar.THURSDAY;
+                    params[8] = SECONDS_PER_DAY * 1000;
+                }
```

Comment:
This actually needs to replace the block above, which is:
~~~
                //endDayOfWeek and endTime workaround                                                                                                       
                if (params[7] == 6 && params[8] == 0 &&
                    (zoneId.equals("Africa/Cairo"))) {
                    params[7] = 5;
                    params[8] = 86400000;
                }
~~~
i.e. the same thing but using hardcoded values rather than constants.

Reply:
Thank you for the review. Removed the dup code mentioned, pushed commit [Code clean up](https://github.com/openjdk/jdk8u-dev/pull/305/commits/90d26a57fe1f67a9efaf1e4b1894f046037bb941)